src/vmprof_common.h

```diff
@@ -32,6 +35,42 @@ typedef struct prof_stacktrace_s {
     void *stack[];
 } prof_stacktrace_s;
 
+#if !DISABLE_GZIP
+static int pipe_gzip(int out_fd, int *gzip_out_fd, int *subproc_pid)
+{
```

Comment:
Well, yeah, why not. I guess it simplifies the code a bit. But it's not much of a difference.